models/iaf_psc_alpha_gap.cpp

```diff
@@ -0,0 +1,551 @@
+  // a double in ms. The grid based iaf_psc_alpha can only handle refractory
+  // periods that are integer multiples of the computation step size (h).
+  // To ensure consistency with the overall simulation scheme such conversion
+  // should be carried out via objects of class nest::Time. The conversion
+  // requires 2 steps:
+  //     1. A time object is constructed defining representation of
+  //        TauR in tics. This representation is then converted to computation
+  //        time steps again by a strategy defined by class nest::Time.
+  //     2. The refractory time in units of steps is read out get_steps(), a
+  //        member function of class nest::Time.
+  //
+  // The definition of the refractory period of the iaf_psc_alpha is consistent
+  // the one of iaf_psc_alpha_ps.
+  //
+  // Choosing a TauR that is not an integer multiple of the computation time
+  // step h will lead to accurate (up to the resolution h) and self-consistent
+  // results. However, a neuron model capable of operating with real valued
+  // spike time may exhibit a different effective refractory time.
+
+  V_.RefractoryCounts_ = Time( Time::ms( P_.TauR_ ) ).get_steps();
+  // since t_ref_ >= 0, this can only fail in error
+  assert( V_.RefractoryCounts_ >= 0 );
+}
+
+/* ----------------------------------------------------------------
+ * Update and spike handling functions
+ */
+
+bool
+iaf_psc_alpha_gap::update_( Time const& origin, const long from, const long to, const bool called_from_wfr_update )
+{
+  assert(
+    to >= 0 && ( delay ) from < kernel().connection_manager.get_min_delay() );
+  assert( from < to );
+
+  const size_t interpolation_order =
+    kernel().simulation_manager.get_wfr_interpolation_order();
+  const double wfr_tol = kernel().simulation_manager.get_wfr_tol();
+  bool wfr_tol_exceeded = false;
+
+  // allocate memory to store the new interpolation coefficients
+  // to be sent by gap event
+  const size_t buffer_size =
+    kernel().connection_manager.get_min_delay() * ( interpolation_order + 1 );
+  std::vector< double > new_coefficients( buffer_size, 0.0 );
+
+  // parameters needed for piecewise interpolation
+  // translation from paper (Hahne et al., 2015):
+  // y_i: V_0
+  // y_ip1: V_1
+  // hf_i: h dot(V_0)
+  // hf_ip1: h dot(V_1)
+  // where 0 denotes before and 1 denotes after one h step
+  double y_i = 0.0, y_ip1 = 0.0, hf_i = 0.0, hf_ip1 = 0.0;
+
+  for ( long lag = from; lag < to; ++lag )
+  {
+
+    if ( called_from_wfr_update )
+    {
+      y_i = S_.y3_;
+      if ( interpolation_order == 3 )
+      {
+        hf_i = V_.h_ * (
+          -S_.y3_ / P_.Tau_
+          + ( P_.I_e_ + S_.y0_ + S_.I_ex_ + S_.I_in_ ) / P_.C_ );
+      }
+    }
+    
+    if ( S_.r_ == 0 )
+    {
+      // neuron not refractory
+      S_.y3_ = V_.P30_ * ( S_.y0_ + P_.I_e_ ) + V_.P31_ex_ * S_.dI_ex_
+        + V_.P32_ex_ * S_.I_ex_ + V_.P31_in_ * S_.dI_in_ + V_.P32_in_ * S_.I_in_
+        + V_.expm1_tau_m_ * S_.y3_ + S_.y3_;
+
```

Comment:
Why is there no `I_gap` term in the dynamics? From what I get from the model code and a small example network this new model calculates the interpolation of the membrane potential and sends the SecondaryEvent, but does not use the incoming information from the gap-junction connected neurons at all? Could you comment on this?

Reply:
we were thinking about introducing this model to have an hpc_benchmark version with gap junctions. however, to build this model would require a lot more thinking and is not connected to 5g development. so we will remove it from this PR again.